sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/InheritedCollidingInterfaceMembers.AfterCSharp8.cs

```diff
@@ -0,0 +1,298 @@
+        }
+    }
+
+    public interface IProtectedMethodSecond
+    {
+        protected string Protected(string value)
+//                       ^^^^^^^^^ Secondary
+        {
+            return value;
+        }
+    }
+
+    public interface IProtectedMethodCommon : IProtectedMethodFirst, IProtectedMethodSecond // Noncompliant {{Rename or add member 'Protected(string)' to this interface to resolve ambiguities.}}
+//                   ^^^^^^^^^^^^^^^^^^^^^^
+    {
+    }
+
+    public interface IPrivateMethodFirst
+    {
+        private string Private(string value)
+        {
+            return value;
+        }
+    }
+
+    public interface IPrivateMethodSecond
+    {
+        private string Private(string value)
+//                     ^^^^^^^ Secondary
+        {
+            return value;
+        }
+    }
+
+    public interface IPrivateMethodCommon : IPrivateMethodFirst, IPrivateMethodSecond // Noncompliant - FP: Issue is raised but since the methods are not accessible on the derived classes or interfaces it doesn't make sense.
+//                   ^^^^^^^^^^^^^^^^^^^^
+    {
+    }
+
+    public interface ISealedMethodFirst
+    {
+        public sealed string Sealed(string value)
+        {
+            return value;
+        }
+    }
+
+    public interface ISealedMethodSecond
+    {
+        public sealed string Sealed(string value)
+//                           ^^^^^^ Secondary
+        {
+            return value;
+        }
+    }
+
+    public interface ISealedMethodCommon : ISealedMethodFirst, ISealedMethodSecond // Noncompliant
+//                   ^^^^^^^^^^^^^^^^^^^
+    {
+    }
+
+    public interface IStaticMethodFirst
+    {
+        public static string Static(string value)
+        {
+            return value;
+        }
+    }
+
+    public interface IStaticMethodSecond
+    {
+        public static string Static(string value)
+//                           ^^^^^^ Secondary
+        {
+            return value;
+        }
+    }
+
+    public interface IStaticMethodDerived : IStaticMethodFirst, IStaticMethodSecond // Noncompliant - FP: static methods are not inherited
+    {
+    }
+
+    public interface IBase1
+    {
+        public static string PublicStaticField;
+    }
+
+    public interface IBase2
+    {
+        public static string PublicStaticField;
+    }
+
+    public interface IDerived : IBase1, IBase2
+    {
+    }
+
+    public interface IIndexer1
+    {
+        public string this[int i]
+        {
+            get { return ""; }
+            set { }
+        }
+    }
+
+    public interface IIndexer2
+    {
+        public string this[int i] // Rule is raised but the message is not clear
+        {
+            get { return ""; } // Noncomliant
+//          ^^^ Secondary
+            set { }
+//          ^^^ Secondary
+        }
+    }
+
+    public interface IIndexerDerived : IIndexer1, IIndexer2 // Noncompliant {{Rename or add members '' and '' to this interface to resolve ambiguities.}}
```

Comment:
does this happen for the class-related rule as well?

Reply:
Good point! I will add a test also for classes.